policy/pa.go

```diff
@@ -288,16 +311,42 @@ func (pa *AuthorityImpl) WillingToIssueWildcard(ident core.AcmeIdentifier) error
 		if baseDomain == icannTLD {
 			return errICANNTLDWildcard
 		}
+		// The base domain can't be in the wildcard exact blacklist
+		if err := pa.checkWildcardHostList(baseDomain); err != nil {
+			return err
+		}
 		// Check that the PA is willing to issue for the base domain
+		// Since the base domain without the "*." may trip the exact hostname policy
+		// blacklist when the "*." is removed we replace it with a single "x"
+		// character to differentiate "*.example.com" from "example.com" for the
+		// exact hostname check.
```

Comment:
Just to immortalize and OOB conversation: this is a bit hacky, and should probably be done in a different way, but doing so would require separating the valid domain structure logic from the policy/blacklist logic. If we want to push that work to another sprint can we add a `TODO` with the relevant ticket.